src/components/dropdownSearch/MultiSelectDropdownSearch/MultiSelectDropdownSearchReducer.tsx

```diff
@@ -0,0 +1,134 @@
+import { DropdownSearchActions, IOptionsDropdownSearchPayload } from '../DropdownSearchActions';
+import {
+  addUniqueSelectedOption,
+  dropdownSearchInitialState,
+  dropdownSearchReducer,
+  getDisplayedOptions,
+  getNextIndexPosition,
+  getOptionsFiltered,
+  IDropdownSearchState,
+  removeSelectedOption,
+} from '../DropdownSearchReducers';
+import { IReduxAction } from '../../../utils/ReduxUtils';
+import {FixedQueue} from '../../../utils/FixedQueue';
+import {IDropdownOption} from '../DropdownSearch';
+import {keyCode} from '../../../utils/InputUtils';
+
+export const multiSelectDropdownSearchReducer = (state: IDropdownSearchState = dropdownSearchInitialState,
+  action: IReduxAction<IOptionsDropdownSearchPayload>) => {
+
+  switch (action.type) {
+    case DropdownSearchActions.addMultiSelect:
+      return {
+        ...state,
+        id: action.payload.id,
+        options: action.payload.optionsDropdown,
+        selectedOptions: new FixedQueue<IDropdownOption>(),
+        filterText: action.payload.filterText,
+        isOpened: false,
+      };
+    case DropdownSearchActions.removeSelectedOption:
+      state.selectedOptions = removeSelectedOption(state, action.payload.selectedOptionValue);
+      state.displayedOptions = getDisplayedOptions(state);
+      return {
+        ...state,
+        id: action.payload.id,
+        selectedOptions: state.selectedOptions,
+        isOpened: false,
+        activeOption: undefined,
+        displayedOptions: state.displayedOptions,
+      };
+    case DropdownSearchActions.removeAllSelectedOptions:
+      state.selectedOptions = new FixedQueue<IDropdownOption>();
+      state.displayedOptions = getDisplayedOptions(state);
+      return {
+        ...state,
+        id: action.payload.id,
+        selectedOptions: state.selectedOptions,
+        displayedOptions: state.displayedOptions,
+      };
+    case DropdownSearchActions.multiSelect:
+      return {
+        ...state,
+        id: action.payload.id,
+        isOpened: true,
+        selectedOptions: state.selectedOptions.push(action.payload.addedSelectedOption),
```

Comment:
It does not.
https://vincent.billey.me/pure-javascript-immutable-array/ "The mutative functions are push, pop, shift, unshift, sort, reverse, splice and delete (kind of)."